networkx/generators/directed.py

```diff
@@ -414,7 +419,6 @@ def sample(v, nodes):
     return G
 
 
-@py_random_state(4)
 @nx._dispatchable(graphs=None, returns_graph=True)
 def random_k_out_graph(n, k, alpha, self_loops=True, seed=None):
```

Comment:
From the perspective of a backend developer, it's nice to have `@py_random_state` or `@np_random_state` _before_ `@nx._dispatchable` so that the `seed` parameter is resolved and the same as used in networkx, which is convenient and can make it easier to match networkx results (but this depends on implementation of any particular function).

What does everyone think about adding e.g. `@np_or_py_random_state(4)` decorator to make it easier for this pattern to be used elsewhere?

Reply:
It might be tricky to make a `@np_or_py_random_state` decorator do what we want with this. The choice of seed object has to be made after checking which code to run. 

Perhaps we could make a decorator specific to this situation... where we know that we will run the numpy code if numpy is installed and the python code if numpy is not installed. Does the name `@np_random_state_if_possible` describe what we want?  Is there a better way to do this? It seems like we are setting ourselves up for making a decorator for each logical choice to be made. (but maybe there aren't many possible variants).